tasks/deploy.py

```diff
@@ -0,0 +1,104 @@
+from . import config
+from .config import Config
+import os
+import invoke
+import getpass
+import subprocess
+from invoke.context import Context
+from typing import Optional, Dict, Any
+import pathlib
+from . import tool
+
+default_public_path_key_name = "ddinfra:aws/defaultPublicKeyPath"
+
+
+def deploy(
+    _: Context,
+    scenario_name: str,
+    stack_name: Optional[str] = None,
+    install_agent: Optional[bool] = None,
+    agent_version: Optional[str] = None,
+    os_type: Optional[str] = None,
+):
+    flags = {}
+
+    if install_agent is None:
+        install_agent = tool.get_default_agent_install()
+    flags["ddagent:deploy"] = install_agent
+
+    os_type = _get_os_type(os_type)
+    flags["ddinfra:osType"] = os_type
+
+    cfg = config.get_config()
+    flags[default_public_path_key_name] = _default_public_path_key_name(cfg, os_type)
+    flags["scenario"] = scenario_name
+    flags["ddagent:version"] = agent_version
+
+    flags["ddinfra:aws/defaultKeyPairName"] = cfg.key_pair
+    flags["ddinfra:env"] = "aws/sandbox"
+
+    if install_agent:
+        flags["ddagent:apiKey"] = _get_api_key()
+
+    _deploy(stack_name, flags)
+
+
+def _get_os_type(os_type: Optional[str]) -> str:
+    os_types = tool.get_os_types()
+    if os_type is None:
+        os_type = tool.get_default_os_type()
+    if os_type not in os_types:
+        raise invoke.Exit(
+            f"the os type '{os_type}' is not supported. Possibles values are {os_types}"
+        )
+    return os_type
+
+
+def _default_public_path_key_name(cfg: Config, os_type: str) -> Optional[str]:
+    if os_type == "Windows" and cfg.defaultPublicKeyPath is None:
+        raise invoke.Exit(
+            f"You must set {default_public_path_key_name} when using this operating system"
+        )
+    return cfg.defaultPublicKeyPath
+
+
+def _deploy(stack_name: Optional[str], flags: Dict[str, Any]) -> None:
+    cmd_args = ["aws-vault", "exec", "sandbox-account-admin", "--", "pulumi", "up"]
```

Comment:
Azure and Google requires login to be done using browser beforehand, so it won't be possible to wrap the command, it will be a pre-comamnd step.
You can also use a pre-defined service account, which is what I do for AWS and injecting `aws-vault` all the time will break that.